### PR TITLE
#918 Fix Hard coded by Http Status code judgment. (EXPOSUREAPP-1929)

### LINKED ISSUE #918: Hard coded by Http Status code judgment
Hard coded by Http Status code judgment.

It is better to use `HttpsURLConnection` to determine the Http Status code.

https://github.com/corona-warn-app/cwa-app-android/blob/4906bd3b16959b1bca336c081a3718ac06f928c4/Corona-Warn-App/src/main/java/de/rki/coronawarnapp/http/HttpErrorParser.kt#L36:L56

---
Internal Tracking ID: EXPOSUREAPP-1929

### PR DESCRIPTION
**Pull request by LeoAndo**
Closes #918 

Fix Hard coded by Http Status code judgment.

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/http/HttpErrorParser.kt [hunk 1]
```diff
@@ -27,30 +27,31 @@ import de.rki.coronawarnapp.exception.http.UnsupportedMediaTypeException
...
+                HttpsURLConnection.HTTP_ACCEPTED -> response
+                HttpsURLConnection.HTTP_NO_CONTENT -> response
+                HttpsURLConnection.HTTP_BAD_REQUEST -> throw BadRequestException()
+                HttpsURLConnection.HTTP_UNAUTHORIZED -> throw UnauthorizedException()
+                HttpsURLConnection.HTTP_FORBIDDEN -> throw ForbiddenException()
+                HttpsURLConnection.HTTP_NOT_FOUND -> throw NotFoundException()
+                HttpsURLConnection.HTTP_CONFLICT -> throw ConflictException()
+                HttpsURLConnection.HTTP_GONE -> throw GoneException()
+                HttpsURLConnection.HTTP_UNSUPPORTED_TYPE -> throw UnsupportedMediaTypeException()
                 429 -> throw TooManyRequestsException()
```
Comment (LeoAndo) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/http/HttpErrorParser.kt:48]: Code not present in HttpsURLConnection.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/http/HttpErrorParser.kt [hunk 2]
```diff
@@ -27,30 +27,31 @@ import de.rki.coronawarnapp.exception.http.UnsupportedMediaTypeException
...
-                505 -> throw HTTPVersionNotSupported()
+                HttpsURLConnection.HTTP_INTERNAL_ERROR -> throw InternalServerErrorException()
+                HttpsURLConnection.HTTP_NOT_IMPLEMENTED -> throw NotImplementedException()
+                HttpsURLConnection.HTTP_BAD_GATEWAY -> throw BadGatewayException()
+                HttpsURLConnection.HTTP_UNAVAILABLE -> throw ServiceUnavailableException()
+                HttpsURLConnection.HTTP_GATEWAY_TIMEOUT -> throw GatewayTimeoutException()
+                HttpsURLConnection.HTTP_VERSION -> throw HTTPVersionNotSupported()
                 511 -> throw NetworkAuthenticationRequiredException()
                 598 -> throw NetworkReadTimeoutException()
                 599 -> throw NetworkConnectTimeoutException()
```
Comment (LeoAndo) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/http/HttpErrorParser.kt:57]: Code not present in HttpsURLConnection.

**Comment by LeoAndo** [comment]
@pwoessner 

please review this PR.

**Comment by tkowark** [comment]
Dear @LeoAndo thank you for your contribution. Please let me quickly remind you of the issue template, though:

> Pull Requests that are not linked to an issue with you as an assignee will be closed, except for minor fixes for typos or grammar mistakes in documentation or code comments.

We'll discuss with the team whether they can review the PR, but next time, please wait for a response in the corresponding issue.

**Comment by LeoAndo** [comment]
@tkowark 

yes, understood. I'll be careful in next time.

**Review by d4rken — APPROVED** [review]
lgtm, thanks for your contribution!

**Pull request merged**